components/net/http_cache.rs

```diff
@@ -857,7 +871,32 @@ impl HttpCache {
                 last_validated: time::now(),
             }),
         };
+        debug!("storing new cached response for {:?}", request.url());
         let entry = self.entries.entry(entry_key).or_insert(vec![]);
+
+        if response
+            .status
+            .as_ref()
+            .map_or(false, |s| s.0 == StatusCode::OK)
+        {
+            // Ensure that any existing complete response is overwritten by the new
```

Comment:
So this is not a good idea I think, since it will make it harder to have a spec compliant cache(needs to support multiple entries, matching on Etag and so on).